src/Microsoft.TemplateEngine.Core/Expressions/MSBuild/MSBuildStyleEvaluatorDefinition.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.TemplateEngine.Core.Contracts;
+using Microsoft.TemplateEngine.Core.Util;
+
+namespace Microsoft.TemplateEngine.Core.Expressions.MSBuild
+{
+    public class MSBuildStyleEvaluatorDefinition
+    {
+        private static readonly Dictionary<Encoding, ITokenTrie> TokenCache = new Dictionary<Encoding, ITokenTrie>();
+
+        private static ITokenTrie GetSymbols(IProcessorState processor)
+        {
+            ITokenTrie tokens;
+            if (!TokenCache.TryGetValue(processor.Encoding, out tokens))
+            {
+                TokenTrie trie = new TokenTrie();
+
+                //Logic
+                trie.AddToken(processor.Encoding.GetBytes("AND"));
+                trie.AddToken(processor.Encoding.GetBytes("OR"));
+                trie.AddToken(processor.Encoding.GetBytes("!"));
+                trie.AddToken(processor.Encoding.GetBytes("&gt;"));
+                trie.AddToken(processor.Encoding.GetBytes("&gt;="));
+                trie.AddToken(processor.Encoding.GetBytes("&lt;"));
+                trie.AddToken(processor.Encoding.GetBytes("&lt;="));
+                trie.AddToken(processor.Encoding.GetBytes("=="));
+                trie.AddToken(processor.Encoding.GetBytes("!="));
+
+                //Braces
+                trie.AddToken(processor.Encoding.GetBytes("("));
+                trie.AddToken(processor.Encoding.GetBytes(")"));
+
+                //Whitespace
+                trie.AddToken(processor.Encoding.GetBytes(" "));
+                trie.AddToken(processor.Encoding.GetBytes("\t"));
+
+                //EOLs
+                trie.AddToken(processor.Encoding.GetBytes("\r\n"));
+                trie.AddToken(processor.Encoding.GetBytes("\n"));
+                trie.AddToken(processor.Encoding.GetBytes("\r"));
+
+                // quotes
+                trie.AddToken(processor.Encoding.GetBytes("'"));
+                TokenCache[processor.Encoding] = tokens = trie;
+            }
+
+            return tokens;
+        }
+
+        private enum Tokens
+        {
```

Comment:
The case for "None" was around the quote matching - in this use case (in the evaluator), the active quoting character is a nullable token that indicates either the character that must be matched to terminate the run or null to indicate that a quoted run is not in progress